lib/plugin/plugin.ts

```diff
@@ -98,4 +98,4 @@ export interface ActionDefinition<T = ContractData> extends Action {
 
 export type ContractBuilder<T = ContractData> =
 	| ContractDefinition<T>
-	| (() => ContractDefinition<T>);
+	| ((mixins?: any) => ContractDefinition<T>);
```

Comment:
@Ereski @edufschmidt Obviously don't want the `any` here which is why this PR is still in draft. I'm holding off until https://github.com/product-os/jellyfish-core/pull/1192 is merged as it looks like you are touching mixins? If not and you already know what the proper type here should be let me know.

Reply:
@joshbwlng I created a `Mixin` type within the core which contains the mixin schema as well as a test that determines whether or not it should be applied to a given contract:

```ts
interface Mixin<TData = ContractData> {
	schema: ContractDefinition<TData>;
	test: (contract: ContractDefinition<TData>) => boolean;
}

export const initialize = <TData = ContractData>(
	contract: ContractDefinition<TData>,
	mixins: Array<Mixin<TData>>,
) {
    ...
}
```

This type is not yet being exported, but I suppose it is what you are looking for, isn't it? If so, let me know and we can export it in the core.

While we wait for a review of the [PR](https://github.com/product-os/jellyfish-core/pull/1192), feel free to suggest any other changes you see fit.

Reply:
@edufschmidt Contracts provided by plugins can no longer be functions as per https://github.com/product-os/jellyfish-worker/pull/1290. This simplified things quite a bit and this is no longer an issue for me. Closing, but thanks for reaching out!